There are two type errors:

src/api.tsx(471,31): error TS6133: 'tagId' is declared but its value is never read.
src/api.tsx(473,3): error TS2740: Type '{}' is missing the following properties from type 'Note': id, note_text, public, created_at, and 3 more.

--- a/src/api.tsx
+++ b/src/api.tsx
@@ -467,8 +467,3 @@ export const getNotes = async (tagId?: string): Promise<Note[]> => {
   if (!response.ok) throw new Error('Failed to fetch notes');
   return await response.json();
 };
-
-export const getNote = async (tagId?: string): Promise<Note> => {
-
-  return {}
-}

--- a/src/components/NoteCard.test.tsx
+++ b/src/components/NoteCard.test.tsx
@@ -14,6 +14,7 @@ vi.mock('./Verse', () => ({
   ),
 }));
 
+window.confirm = vi.fn()
 vi.mock('../api', () => ({
   deleteNote: vi.fn()
 }))

--- a/src/components/NoteCard.tsx
+++ b/src/components/NoteCard.tsx
@@ -26,6 +26,8 @@ const NoteCard = ({ note, onViewInBible, onEdit }: NoteCardProps) => {
   const handleDelete = async (evt: MouseEvent<HTMLButtonElement>) => {
     evt.preventDefault();
     if(note.id) {
+      if(window.confirm('Are you sure you want to delete this note?')) 
+        return;  
       await deleteNote(note.id);
       fetchNotes()
     }

